Advice and resources for junior developers

rdormer opened the pull request:
This wound up being less about technical specifics and more about general mindset.  Let me know if it's better this way, or if we'd rather dive into technical things instead.

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+[A Taxonomy for "Bad Code
+Smells"](http://mikamantyla.eu/BadCodeSmellsTaxonomy.html)
+[The Law of Leaky
+Abstractions](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/)
```
@rdormer  Looks like these links are being displayed one next to the other. Can you make them look like a list?

lubc replied:
[](url)
<img width="1314" alt="Screen Shot 2020-03-05 at 12 46 18 PM" src="https://user-images.githubusercontent.com/7442887/76004091-59785e00-5edf-11ea-9c22-b809f9942c6c.png">

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+junior developers are scared of making mistakes - imagine their
+surprise when they find themselves working on something that was
+written by another junior developer who did, in fact, make some major
+mistakes - mistakes that are now their responsibility to fix.
```
@rdormer You're missing the `<!--more-->` tag. We need it so it doesn't display the entire article in the index page. Maybe you can break this first paragraph with it.

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+(communication solves a lot of problems up front!)
+[The Developer’s Edge: How To Become A Senior
+Developer](https://medium.com/zerotomastery/developers-edge-how-to-become-a-senior-developer-f1ec1738cf45)
+ (notice how few of these skills are technical!)
```
@rdormer Same here. Can you format this one below the other?

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+---
+layout: post
+title: "Some Resources and Advice For Junior Developers"
+date: 2020-03-02 14:30:00
```
@rdormer I can publish this today or tomorrow

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+[Beautiful Code](https://alarmingdevelopment.org/?p=79)
+[You're a Bad Programmer. Embrace
+It.](https://dzone.com/articles/youre-bad-programmer-embrace)
+[Teach Yourself Programming in Ten Years](http://www.norvig.com/21-days.html)
```
@rdormer Same about these links

lubc replied:
@rdormer It would be nice to have some kind of conclusion or recap at the end of the article

lubc commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+layout: post
+title: "Some Resources and Advice For Junior Developers"
+date: 2020-03-02 14:30:00
+categories: []
```
```suggestion
categories: ["learning",  "software-development"]
```

@rdormer Let's add these categories

FionaDL commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+it explains almost everything you observe as a working developer.
+Because perfection isn't profitable, real world applications are
+developed under less than ideal circumstances, often under intense
+pressure, and with inevitable compromises.  Large applications sprawl,
```
It looks like there are three sentences above with double spaces after the "."

FionaDL commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+get messy, and develop stagnant corners and scary basements.
+Questionable or just plain puzzling decisions are made by developers
+without explanation, justification, or documentation by developers who
+then leave the company, never to be seen or heard from again.  Most
```
Here too.

FionaDL commented on _posts/2020-03-02-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,117 @@
+the obvious problems that this can cause with teamwork and
+professional behavior, losing perspective in this way can lead to
+skills starting to stagnate. When you take on a job as a developer,
+wether you realize it or not, you're implicitly taking on another
```
I think it's "whether" ?

etagwerker commented on _posts/2020-03-12-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,121 @@
+expect a lot of transparency in return. And that expectation will
+only grow as you acquire more and more experience. Your experience
+will have value, and your employers will expect you to share that
+value with those around you, *without* stepping on any toes in the
```
```suggestion
value with those around you by coaching and mentoring teammates, *without* stepping on any toes in the
```

etagwerker commented on _posts/2020-03-12-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,121 @@
+Developer](https://medium.com/zerotomastery/developers-edge-how-to-become-a-senior-developer-f1ec1738cf45)
+ (notice how few of these skills are technical!)
+
+# Stay humble
```
```suggestion
# Stay Humble
```

rdormer commented:
@etagwerker @lubc I think I've addressed all of the review items at this point, and bumped the date to tomorrow.  Feel free to publish whenever you're ready!

etagwerker commented on _posts/2020-03-17-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,121 @@
+
+# Tough Love
+
+One of the first things that a journeyman programmer will have to
```
```suggestion
One of the first things that a journeyperson programmer will have to
```

etagwerker commented on _posts/2020-03-17-junior-dev-advice.markdown:
```diff
@@ -0,0 +1,121 @@
+was incomplete or mistaken. Recognize that no matter how much you've
+learned, it's only the base for the next thing that you're going to
+learn. And realize that no matter how far you've come, you can still
+make dumb mistakes - and program accordingly. In a field full of big
```
@rdormer In fact, I like to say that "if you are not making mistakes, you are not learning fast enough."

You can take it or leave it. :)
